what-is-chalk.mdx

```diff
@@ -18,215 +18,481 @@ import { PyDiffEditor } from '@/components/Editor'
+- Pre-warm caches and backfill pipelines for performance
 
-Add a caching policy with one line of code in your feature definition:
+Add a caching policy with one line of code in our feature definition:
 
 <PyDiffEditor html={addCachingCode} />
 
-Optionally warm feature caches by executing resolvers on a schedule:
 
-```sql
--- resolves: User
--- source: snowflake
--- cron: 1h
-select
-    user_id as id,
-    credit_report,
-    ts
-from credit_scores
+### How do you deploy and query features with Chalk?
+
+After defining pipelines, deploy features to a branch in < 100ms with:
+Chalk automatically detects and uses your current Git branch name without requiring you to explicitly specify it:
+
+```shell
+chalk apply --branch
 ```
 
-Or override staleness tolerances at query time when you need fresher
-data for your models:
+A successful execution of the `chalk apply --branch` command creates a new isolated branch (in this case, 'fraud-model-v2').
+Chalk scans through resolvers, lints features, and deploys them to this separate branch.
+This isolation enables testing changes independently before merging them into the production environment.
 
-```py
-from chalk.client import ChalkClient
+```
+✓ Found resolvers
+✓ Deployed to branch 'fraud-model-v2'
+```
 
-ChalkClient().query(
-    input={
-        User.id: 123,
-    },
-    output=[User.credit_report],
-    max_staleness={User.credit_report: "1m"}
-)
+Query for features against this branch and perform QA workflows on branch deployments to ensure pipelines perform as expected:
+
+```shell
+chalk query --branch --in user.id=47
 ```
 
-### Batch ETL ingestion
+This command returns the user's features from your branch deployment, showing all computed fields for the specified `user.id`.
 
-Chalk also makes it simple to generate training sets from data warehouse
-sources -- join data from services like S3, Redshift, BQ, Snowflake
-or other custom sources with historical features computed online.
-Specify a cron schedule on an `@offline` resolver, and Chalk will automatically ingest
-data with support for incremental reads:
+```
+Using '--out=user'
+Results
+https://chalk.ai/environments/devx/query-runs/4701020347
+Branch: elvis
+Environment: devx
+
+ Name                                             Hit?  Value
+─────────────────────────────────────────────────────────────────────────────────────────────────────
+ user.average_rating_given                  4.3125
+ user.birthday                              "1987-12-28"
+ user.created_at                            "2024-08-07T20:54:42.942294+00:00"
+ user.email                                 "98178ad58fe8481db74996996d6e8de7@google.ru"
+ user.first_name                            "Marseda"
+ user.id                                    47
+ user.last_name                             "Karkaletsis"
+ user.review_count                          32
+ user.total_orders_placed                   48
+ user.unique_products_inquired_about        203
+```
```

Comment:
can we collapse this into the shell output all together?

Reply:
not sure what you mean

Reply:
do you mean having an accordion component so the code block can be minimized and expanded?